agent/agent.go

```diff
@@ -156,6 +164,54 @@ func (a *agent) runLoop(ctx context.Context) {
 	}
 }
 
+// reportLifecycleLoop reports the current lifecycle state once.
+// Only the latest state is reported, intermediate states may be
+// lost if the agent can't communicate with the API.
+func (a *agent) reportLifecycleLoop(ctx context.Context) {
```

Comment:
I like how this loop is done. Retrying and fetching the new state in the loop is cool.